src/GovUk.Education.ExploreEducationStatistics.Data.Model/Services/Interfaces/IDataService.cs

```diff
@@ -11,9 +11,17 @@ public interface IDataService<TEntity> where TEntity : class
 
         int Count(Expression<Func<TEntity, bool>> expression);
 
+        IEnumerable<TEntity> All();
+
+        TEntity Find(object id);
```

Comment:
Not super knowledgeable Entity Framework, but shouldn't we be using type parameters to define what the ID types should in this interface?

It feels to me (with my rudimentary C# knowledge) that this would be fine:

```
public interface IDataService<TEntity, Key> where TEntity : class
{
    TEntity Find(Key id);
}
```

Reply:
Thanks, I've added the Key type parameter in.